File: Solutions/Length_of_Last_Word/Length_of_Last_Word.go
package main

func lengthOfLastWord(s string) int {
	var res int
	i := len(s) - 1
	for ; i >= 0 && s[i] == ' '; i-- {

	}
	for ; i >= 0; i-- {
		if s[i] != ' ' {
			res++
		} else {
			break
		}
	}
	return res
}

File: Solutions/Length_of_Last_Word/Length_of_Last_Word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int A = 0;
        for (int i=s.size()-1; i>=0; --i) {
            if (s[i] != ' ') ++A;
            else if (A) return A;
        }
        return A;
    }
};

//Fastest
class Solution {
public:
	int lengthOfLastWord(string s) {
		vector<string> temp;
		auto i = s.begin(), j = s.end();
		j--;
		int res = 0;
		while ((*i) == ' ') i++;
		while ((*j) == ' ') j--;
		while (i <= j) {
			if ((*i) == ' ') {
				res = 0;
			}else
			{
				res++;
			}
			i++;
		}
		return res;
	}
};